src/main/java/space/npstr/magma/Magma.java

```diff
@@ -58,6 +61,9 @@
     private static final int DEFAULT_POOL_BUFFER_SIZE = 16 * 1024;
 
     private final FluxSink<LifecycleEvent> lifecycleSink;
+    @Nullable
```

Comment:
Can't we avoid this (misleading) annotation by moving the initialization of those two fields into the constructor?

Reply:
Then we would have to subscribe to the Flux. I'm not sure if subscribing prematurely is bad practice in our particular case.